### DISCUSSION TITLE
Checking if the Map is available

### DISCUSSION THREAD

**Discussion by tbenci** [visgl/react-google-maps#589] [Q&A]
Howdy,
Is there a way to be able to check whether the APIProvider has loaded successfully and that calling a MAP or any other service (places etc.) is going to work.

For example, If I deliberately use an incorrect API key (or remove my localhost:3000 from the filters on Google Dashboard) the hooks return  apiIsLoaded true and apiStatus LOADED yet the Google Maps JavaScript API error: InvalidKeyMapError pops up and any attempt to display the map or call a places action throws errors.

I'd like to be able to test for an correct loading of the map services BEFORE I call them?

**Comment by usefulthink** [comment]
That might be a problem because - as far as I know - the Maps API isn't running the key-validation until it first triggers a billable event. 
iirc, it might not even happen directly when creating a map instance if the map isn't in the viewport.

The only way to get information about whether the API key is valid is through the `gm_authFailure` callback, and [we do register a listener for that](https://github.com/visgl/react-google-maps/blob/main/src/libraries/google-maps-api-loader.ts#L166-L169), so you should be able to see that with the `useApiLoadingStatus()` hook.
